### PR TITLE
release 0.7.5 updates

### PR DESCRIPTION
**Pull request by ericl**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.

### CODE REVIEW

File: ci/stress_tests/application_cluster_template.yaml [hunk 1]
```diff
@@ -33,11 +33,8 @@ idle_timeout_minutes: 5
 # Cloud-provider specific configuration.
 provider:
     type: aws
-    region: us-west-2
-    # Availability zone(s), comma-separated, that nodes may be launched in.
-    # Nodes are currently spread between zones by a round-robin approach,
-    # however this implementation detail should not be relied upon.
-    availability_zone: us-west-2b
+    region: us-east-1
+    availability_zone: us-east-1a
```
Comment (robertnishihara) [inline ci/stress_tests/application_cluster_template.yaml]: us-west-2 is recommended by AWS for people who live in California and is the region we've standardized around in almost all of our config files, so let's keep that. This also means reverting the AMI IDs.

File: ci/stress_tests/application_cluster_template.yaml [hunk 2]
```diff
@@ -69,11 +66,11 @@ head_node:
...
 worker_nodes:
     InstanceType: <<<WORKER_TYPE>>>
-    ImageId: ami-0027dfad6168539c7  # Amazon Deep Learning AMI (Ubuntu), Version 21.2
+    ImageId: ami-0757fc5a639fe7666
 
     # Run workers on spot by default. Comment this out to use on-demand.
-    InstanceMarketOptions:
-        MarketType: spot
+#    InstanceMarketOptions:
+#        MarketType: spot
```
Comment (robertnishihara) [inline ci/stress_tests/application_cluster_template.yaml]: Why this change?

File: ci/stress_tests/application_cluster_template.yaml [hunk 3]
```diff
@@ -90,10 +87,9 @@ file_mounts: {
 # List of shell commands to run to set up nodes.
 setup_commands:
     - echo 'export PATH="$HOME/anaconda3/envs/tensorflow_<<<PYTHON_VERSION>>>/bin:$PATH"' >> ~/.bashrc
-    - ray || wget https://s3-us-west-2.amazonaws.com/ray-wheels/releases/<<<RAY_VERSION>>>/<<<RAY_COMMIT>>>/ray-<<<RAY_VERSION>>>-<<<WHEEL_STR>>>-manylinux1_x86_64.whl
-    - rllib || pip install -U ray-<<<RAY_VERSION>>>-<<<WHEEL_STR>>>-manylinux1_x86_64.whl[rllib]
-    - pip install tensorflow-gpu==1.12.0
-    - echo "sudo halt" | at now + 60 minutes
+    - wget --quiet https://s3-us-west-2.amazonaws.com/ray-wheels/releases/<<<RAY_VERSION>>>/<<<RAY_COMMIT>>>/ray-<<<RAY_VERSION>>>-<<<WHEEL_STR>>>-manylinux1_x86_64.whl
+    - source activate tensorflow_p36 && pip install -U ray-<<<RAY_VERSION>>>-<<<WHEEL_STR>>>-manylinux1_x86_64.whl[rllib]
```
Comment (robertnishihara) [inline ci/stress_tests/application_cluster_template.yaml]: How did this work before without the source activate? Should we just put the source activate in the bashrc instead of the export command above?

File: doc/dev/RELEASE_PROCESS.rst [hunk 4]
```diff
@@ -3,12 +3,19 @@ Release Process
 
 This document describes the process for creating new releases.
 
-1. **Increment the Python version:** Create a PR that increments the Python
-   package version. See `this example`_.
+3. **Create a release branch:** Create the branch from the desired commit on master
```
Comment (robertnishihara) [inline doc/dev/RELEASE_PROCESS.rst]:
```suggestion
1. **Create a release branch:** Create the branch from the desired commit on master
```

File: ci/stress_tests/application_cluster_template.yaml (hunk 1 above)
Comment (robertnishihara) [reply ci/stress_tests/application_cluster_template.yaml]: Same with stress_testing_config.yaml

File: ci/stress_tests/run_application_stress_tests.sh [hunk 5]
```diff
@@ -74,9 +74,9 @@ test_impala(){
                 s/<<<RAY_COMMIT>>>/$RAY_COMMIT/;
                 s/<<<CLUSTER_NAME>>>/$TEST_NAME/;
                 s/<<<HEAD_TYPE>>>/p3.16xlarge/;
-                s/<<<WORKER_TYPE>>>/m5.24xlarge/;
-                s/<<<MIN_WORKERS>>>/5/;
-                s/<<<MAX_WORKERS>>>/5/;
+                s/<<<WORKER_TYPE>>>/m4.16xlarge/;
+                s/<<<MIN_WORKERS>>>/9/;
+                s/<<<MAX_WORKERS>>>/9/;
```
Comment (robertnishihara) [inline ci/stress_tests/run_application_stress_tests.sh]: Why change the instance type and number of machines? Or is it a question of what your instance limits were?

File: ci/stress_tests/run_application_stress_tests.sh [hunk 6]
```diff
@@ -85,10 +85,11 @@ test_impala(){
             RLLIB_DIR=../../python/ray/rllib/
             ray --logging-level=DEBUG up -y "$CLUSTER" &&
             ray rsync_up "$CLUSTER" $RLLIB_DIR/tuned_examples/ tuned_examples/ &&
-            sleep 1 &&
-            ray --logging-level=DEBUG exec "$CLUSTER" "rllib || true" &&
-            ray --logging-level=DEBUG exec "$CLUSTER" "
-                rllib train -f tuned_examples/atari-impala-large.yaml --ray-address='localhost:6379' --queue-trials" &&
+            # HACK: the test will deadlock if it scales up slowly, so we have to wait
```
Comment (robertnishihara) [inline ci/stress_tests/run_application_stress_tests.sh]: Any idea why we didn't run into this before?

File: ci/stress_tests/stress_testing_config.yaml [hunk 7]
```diff
@@ -86,14 +86,14 @@ file_mounts: {
 # List of shell commands to run to set up nodes.
 setup_commands:
     # Consider uncommenting these if you run into dpkg locking issues
-    # - sudo pkill -9 apt-get || true
-    # - sudo pkill -9 dpkg || true
-    # - sudo dpkg --configure -a
+    - sudo pkill -9 apt-get || true
```
Comment (robertnishihara) [inline ci/stress_tests/stress_testing_config.yaml]: Did you run into this issue? I haven't run into it before when stress testing.

File: doc/dev/RELEASE_PROCESS.rst [hunk 8]
```diff
@@ -44,6 +45,17 @@ This document describes the process for creating new releases.
...
+
+   .. code-block:: bash
+
+       ray/ci/long_running_tests/start_workloads.sh
+
+   You can use the `check_workloads.sh` script to verify the workloads are running.
+   Let them run for at least 24 hours, and check them again. They should all still
+   be running (printing new iterations), and their CPU load should be stable when
+   you view them in the AWS monitoring console (not increasing over time).
+
```
Comment (robertnishihara) [inline doc/dev/RELEASE_PROCESS.rst]: Do you also want to add something about publishing performance numbers?

File: ci/stress_tests/application_cluster_template.yaml (hunk 1 above)
Comment (ericl) [reply ci/stress_tests/application_cluster_template.yaml]: This is actually a serious problem for these YAML files. The fact is that with high probability, the user is not going to be in the right region.

This is going to need a long-term solution.

File: ci/stress_tests/application_cluster_template.yaml (hunk 2 above)
Comment (ericl) [reply ci/stress_tests/application_cluster_template.yaml]: Spot instances (1) may not launch immediately, (2) may be killed at any time. These are both generally bad if launching a small number of instances.

They do have an advantage in that you can launch more without requiring a limit increase, however here what happens is that AWS just sometimes won't fulfill your spot requests if try to request this many without a quota anyways...

File: ci/stress_tests/application_cluster_template.yaml (hunk 3 above)
Comment (ericl) [reply ci/stress_tests/application_cluster_template.yaml]: This is switching to the DLAMI, which is the better AMI (tm).

File: ci/stress_tests/run_application_stress_tests.sh (hunk 5 above)
Comment (ericl) [reply ci/stress_tests/run_application_stress_tests.sh]: More people have m4 quota than m5 quota, which is rather rare.

File: ci/stress_tests/run_application_stress_tests.sh (hunk 6 above)
Comment (ericl) [reply ci/stress_tests/run_application_stress_tests.sh]: I think we did, but just retried until it got lucky.

File: ci/stress_tests/stress_testing_config.yaml (hunk 7 above)
Comment (ericl) [reply ci/stress_tests/stress_testing_config.yaml]: It might be the DLAMI switch.

File: doc/dev/RELEASE_PROCESS.rst (hunk 8 above)
Comment (ericl) [reply doc/dev/RELEASE_PROCESS.rst]: Let's do that separately.

File: ci/stress_tests/application_cluster_template.yaml (hunk 1 above)
Comment (pcmoritz) [reply ci/stress_tests/application_cluster_template.yaml]: We can use something like `aws ec2 describe-images --owners amazon` to get the right image ID I think. Unfortunately out of the box this is quite slow, and requires the right region to be set up in `aws configure` but there should be workarounds to do it. We might be able to make it faster by using some of the flags in https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/finding-an-ami.html or caching the information.

File: ci/stress_tests/application_cluster_template.yaml (hunk 3 above)
Comment (robertnishihara) [reply ci/stress_tests/application_cluster_template.yaml]: From the diff, it looks like it was already using the DLAMI.

File: ci/stress_tests/application_cluster_template.yaml (hunk 1 above)
Comment (ericl) [reply ci/stress_tests/application_cluster_template.yaml]: How about we switch it to us-east for now, so that the checked in version always reflects the exact configuration that was tested. The next release volunteer can feel free to set it to their personal preference for whatever region they have.

Comment (robertnishihara) [reply ci/stress_tests/application_cluster_template.yaml]: True, though eventually these tests will presumably be primarily run in an automated fashion using only a single account, at which point it will be less of an issue.

File: ci/stress_tests/application_cluster_template.yaml (hunk 3 above)
Comment (ericl) [reply ci/stress_tests/application_cluster_template.yaml]: Ah, I see it's due to the export line above. That's bad practice, to inject lines, since it is implicitly changing the environment by side-effect.

File: ci/stress_tests/application_cluster_template.yaml (hunk 1 above)
Comment (robertnishihara) [reply ci/stress_tests/application_cluster_template.yaml]: that proposal is fine with me

**Review by pcmoritz — APPROVED** [review]
LGTM

File: python/ray/scripts/scripts.py [hunk 9]
```diff
@@ -762,6 +762,12 @@ def stack():
     subprocess.call(COMMAND, shell=True)
 
 
+@cli.command()
+def microbenchmark():
+    from ray.ray_perf import main
```
Comment (richardliaw) [inline python/ray/scripts/scripts.py]: Can you add some docstring here?
